doc: nrf_modem: document how to calculate 1PPS pulse time (#896)
nrfconnect/sdk-nrfxlib: nrf_modem/doc/gnss_interface.rst

   1PPS is only supported by modem firmware v1.3.0 or later.

Resolving the UTC time of 1PPS pulse occurrence
===============================================

> As the next pulse time (beginning of an UTC second) is calculated from the previous valid PVT fix, the latest PVT fix notification needs to be used when resolving the UTC time of the 1PPS pulse.

Reviewer: ```suggestion
As the time pulse (aligned to the top of an UTC second) is calculated from the previous valid PVT fix, the latest PVT fix notification needs to be used when resolving the UTC time of the 1PPS pulse.
```

Author: Or maybe "time of the pulse"?

Can you say "aligned to the **top** of a second"? @peknis?

Reviewer 2: I have to say it sounds a bit weird. Can we somehow go around this?

Author: "aligned to the beginning of", "aligned to the boundary of". Any other ideas?

Reviewer: This top of the second thing is basically reflecting u-blox timing parlance. I always assumed it is correct way to say it, but I don't know if I've actually seen it being used elesewhere.

Reviewer: Okay, top of a second seems to be common expression in the domain of timing. If you google "top of a second" & UTC you'll find plenty of hits

Author: Seems that you are right, this just didn't ring a bell for me. @peknis, is "aligned to the top of an UTC second" then fine for you?

And do we go with "the time pulse"? To me "time of the pulse" sounds better, because that's what we're actually calculating.

Reviewer 2: To be honest, I'm not really familiar with the topic. If you say the "top of a second" is a common expression, so be it. Same with the "time of the pulse".

> While the 1PPS pulse does not have any time delay, the PVT fix notification will always have some delay, both from the PVT solution calculation (approximately 100ms) and the notification message delivery.

Reviewer 2: ```suggestion
While the 1PPS pulse does not have any time delay, the PVT fix notification will always have some delay, both from the PVT solution calculation (approximately 100 ms) and the notification message delivery.
```

Therefore, the 1PPS pulse may come before the PVT notification that was used to calculate the exact pulse time.

The UTC time of the 1PPS pulse can be calculated using the following formula:

.. math::

   round\ to\ nearest\ second\ (t_{(PVT,GPST)}+∆t+100 ms)

> t\ :sub:`(PVT,GPST)` \ is the GPS time stamp in the previous PVT notification and |delta| t is the time difference between the 1PPS pulse and the reception of previous PVT notification:

Reviewer 2: ```suggestion
t\ :sub:`(PVT,GPST)` \ is the GPS time stamp in the previous PVT notification and |delta| ``t`` is the time difference between the 1PPS pulse and the reception of previous PVT notification:
```

.. |delta| unicode:: 0x394 .. capital delta sign
   :rtrim:

.. math::

   ∆t=t_P-t_{PVT}

> Thus, |delta| t is always positive.

Reviewer 2: ```suggestion
Thus, |delta| ``t`` is always positive.
```

Author: @peknis This gets rendered like this, is this what we want?

![image](https://user-images.githubusercontent.com/2588548/211296046-c29a8ee4-1f22-450b-be56-917875d8f6e2.png)

Reviewer 2: I thought we might want "t" to stand out. If it bothers you, take it out. It is not code, after all.

Author: I think it looks better if both `delta` and `t` are the same color and font. I tried the one below, but that gets rendered like this:
![image](https://user-images.githubusercontent.com/2588548/211302430-942a9a1e-fcba-4b57-b4eb-3eb961c7f677.png)
```suggestion
Thus, ``|delta| t`` is always positive.
```
So maybe we'll go with the original...
